Source/extensions/processcontainers/ContainerProducer.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2024 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#pragma once
+
+#include "Module.h"
+#include "IProcessContainers.h"
+#include "ContainerAdministrator.h"
+
+namespace Thunder {
+namespace ProcessContainers {
+
+    template<typename PRODUCER, const IContainer::containertype CONTAINERTYPE>
+    class ContainerProducerRegistrationType {
+    public:
+        ContainerProducerRegistrationType(const ContainerProducerRegistrationType&) = delete;
+        ContainerProducerRegistrationType(ContainerProducerRegistrationType&&) = delete;
+        ContainerProducerRegistrationType& operator=(const ContainerProducerRegistrationType&) = delete;
+        ContainerProducerRegistrationType& operator=(ContainerProducerRegistrationType&&) = delete;
+
+        ContainerProducerRegistrationType()
+        {
+           ContainerAdministrator::Instance().Announce<PRODUCER>(CONTAINERTYPE);
+        }
+
+        ~ContainerProducerRegistrationType()
+        {
+            ContainerAdministrator::Instance().Revoke(CONTAINERTYPE);
+        }
```

Comment:
Why not have in this class, the PRODUCER an aggregated member and than announce (CONTAINERTYPE, IContainerProducer& _Producer); Saves you a list with autopointers and new in the administrator and lifetime is guaranteed aligned with the Announce and Revoke. 
It is my allergy to visible "new" 's and std::unique_ptr that triggered this deep dive ;-) and potential simple fix to avoid them..